app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb

```diff
@@ -11,7 +11,8 @@ module AuthenticateIdToken
       ) do
 
         def call
-          decode_id_token
+          discovered_provider
+          decoded_id_token
```

Comment:
Are you calling these methods early to make the `call` method look like an index of what's happening? I wouldn't bother doing so in this situation. It's forcing you to memoize an object that you could just be returning (`@discovered_provider`) and reads a bit strangely since these two methods are essentially 'getters'.  You could reword the methods to be a bit more action oriented like `discover_provider` and `decode_id_token` perhaps. Otherwise, I would just remove them and let them happen as helpers in `verify_id_token`.

Reply:
yes, that was the idea. would you rather have something like this?

```
def decode_id_token
  decoded_id_token
end
```
?

Reply:
Hmm good question. @jonahx what's the "command class" philosophy here? It seems odd to be calling helper methods early just to produce a readable index. Should they be removed? Or is there a bigger restructure that needs to happen here to create a `call` method with meaningful line items?